### PR TITLE
Chart: Add `CiliumNetworkPolicy`.

### PR DESCRIPTION
**Pull request by whites11**
When switching to cilium, we lost support for defining network policies that define CIDRs as targets, such as the one we use in this app to allow egress to the API server.
This PR adds a ciliumnetworkpolicy (disabled by default, enabled by `cluster-operator` in vintage clusters) to overcome this cilium limitation.

See https://github.com/giantswarm/giantswarm/issues/23014#issuecomment-1476207008

### CODE REVIEW

**Comment by whites11** [comment]
@Gacko what's the status here?

**Comment by Gacko** [comment]
Merge as long as it works, I'll put my minor rework on your changes.

**Review by Gacko — APPROVED** [review]

**Comment by Gacko** [comment]
But please recover the `values.schema.json` before and only change the part related to the `CilliumNetworkPolicy`. That would be great!

File: helm/cert-manager-app/charts/cert-manager-giantswarm-clusterissuer/values.yaml [hunk 1]
```diff
@@ -20,3 +20,4 @@ resources:
   requests:
     cpu: 100m
     memory: 150Mi
+
```
[the remark this answers is not in the document]
Comment (whites11) [reply helm/cert-manager-app/charts/cert-manager-giantswarm-clusterissuer/values.yaml]: I guess IDE changes

File: helm/cert-manager-app/values.schema.json
[the remark this answers is not in the document]
Comment (whites11) [reply helm/cert-manager-app/values.schema.json:1]: done

**Pull request merged**